airbyte-integrations/connectors/source-chargebee/source_chargebee/streams.py

```diff
@@ -132,6 +117,77 @@ def read_records(
         # Always return an empty generator just in case no records were ever yielded
         yield from []
 
+
+class SemiIncrementalChargebeeStream(ChargebeeStream):
+    cursor_field = "updated_at"
+
+    def __init__(self, start_date: str):
+        # Convert `start_date` to timestamp(UTC).
+        self._start_date = pendulum.parse(start_date).int_timestamp if start_date else None
+
+    def get_starting_point(self, stream_state: Mapping[str, Any], item_id: str) -> str:
+        start_point = self._start_date
+
+        if stream_state and stream_state.get(item_id, {}).get(self.cursor_field):
+            start_point = max(start_point, stream_state[item_id][self.cursor_field])
+
+        return start_point
+
+    def parse_response(
+        self, list_result: ListResult, stream_slice: Mapping[str, Any] = None, stream_state: Mapping[str, Any] = None
+    ) -> Iterable[Mapping]:
+        starting_point = self.get_starting_point(stream_state=stream_state, item_id=stream_slice["item_id"])
+
+        for message in list_result:
+            record = message._response[self.name]
+            if record[self.cursor_field] > starting_point:
+                yield record
+
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]):
+        """
+        Override airbyte_cdk Stream's `get_updated_state` method to get the latest Chargebee stream state.
+        """
+        item_id = latest_record["parent_item_id"]
+        latest_cursor_value = latest_record.get(self.cursor_field)
+        current_stream_state = current_stream_state.copy()
+        current_state = current_stream_state.get(item_id)
+        if current_state:
+            current_state = current_state.get(self.cursor_field)
+
+        current_state_value = current_state or latest_cursor_value
+        max_value = max(current_state_value, latest_cursor_value)
+        current_stream_state[item_id] = {self.cursor_field: max_value}
+        return current_stream_state
+
+
+class IncrementalChargebeeStream(SemiIncrementalChargebeeStream):
+    include_deleted = "true"
+
+    @property
+    def state_checkpoint_interval(self) -> int:
```

Comment:
this can be `state_checkpoint_interval = self.page_size`

Reply:
`page_size` could be changed in each stream class, so we need to use `self.page_size`. And we can't use `self` in class attributes because there is no thing as `self`. So this means that we can't write smth like this:
```
class IncrementalChargebeeStream(...):
    state_checkpoint_interval = self.page_size
```

Reply:
Thanks for the explanation!